custom/enikshay/expressions.py

```diff
@@ -153,38 +150,22 @@ def __call__(self, item, context=None):
         return None
 
     @staticmethod
+    @ucr_context_cache(vary_on=('person_id',))
     def _get_referral(context, domain, person_id):
```

Comment:
any chance that by not including `domain` in the cache key, you're opening up a attack vector of accessing another domain's data through the cache? (i.e. I can make a request the involves looking up a person_id, and it'll look it up in the cache without checking it's for the right domain?) Or maybe that's not a problem because the `context` variable is going to be limited to a scope smaller than domains anyway? Any downside to varying on `domain` as well? Seems overall more correct to me.

Reply:
I think it's fine to vary on domain, but you're correct that context is only valid in a very short scope (only the current document change, it's just a local dict, not open to other processes)

Reply:
I don't feel strongly about it, was mostly just trying to mimic the existing functionality.  Like J$ said, the context is specific to a document change, so it's already domain-scoped and most of these functions get the domain _from_ the context itself, so it'd be redundant to specify the domain.  Setting the cache would essentially be:
```python
context.cache[(context.domain, mykey)] = get_my_value(context)
```
I suppose `ucr_context_cache` could assert that the context has a domain or something like that.

Reply:
Actually, as I think about it more, it seems reasonable to add `domain` as a required parameter for `EvaluationContext`.  We pretty consistently assume that these contexts exist within the scope of a domain (including in this PR), so it doesn't seem unreasonable to require that explicitly.  I do think that's out of scope for this PR, but I could be talked into trying that out separately.

Reply:
I'd encourage it if you have some good ideas, but yeah this PR certainly doesn't need to block on it :)